### PR TITLE
DOC: Fix malformed docstrings in ma.

### PR DESCRIPTION
**Pull request by takanori-pskq**
Fixing documents for some functions in `numpy.ma`(e.g. `numpy.ma.vstack`, https://numpy.org/devdocs/reference/generated/numpy.ma.vstack.html ). The documents is malformed because the indent of notes section is wrong.

### CODE REVIEW

File: numpy/ma/extras.py [hunk 1]
```diff
@@ -258,7 +258,8 @@ def getdoc(self):
         doc = getattr(npfunc, '__doc__', None)
         if doc:
             sig = self.__name__ + ma.get_object_signature(npfunc)
-            locdoc = "Notes\n-----\nThe function is applied to both the _data"\
+            locdoc = "\n\n        Notes\n        -----\n        "\
+                     "The function is applied to both the _data"\
                      " and the _mask, if any."
```
Comment (eric-wieser) [inline numpy/ma/extras.py:263]: Perhaps instead we should use `inspect.cleandoc(doc)` before concatenating?

Comment (takanori-pskq) [reply numpy/ma/extras.py]: @eric-wieser Thanks for reviewing. I think that's right, so I fixed the code.

File: numpy/ma/extras.py [hunk 2]
```diff
@@ -257,7 +258,8 @@ def getdoc(self):
         npfunc = getattr(np, self.__name__, None)
         doc = getattr(npfunc, '__doc__', None)
         if doc:
-            sig = self.__name__ + ma.get_object_signature(npfunc)
+            sig = self.__name__ + ma.get_object_signature(npfunc) + '\n'
+            doc = inspect.cleandoc(doc)
             locdoc = "Notes\n-----\nThe function is applied to both the _data"\
                      " and the _mask, if any."
```
Comment (rossbar) [inline numpy/ma/extras.py:264]:
```suggestion
            locdoc = ("Notes\n-----\nThe function is applied to both the _data"
                      " and the _mask, if any.")
```
Might as well take the opportunity to clean up the style here a bit as well.

**Review by rossbar — COMMENTED** [review]
Applying `cleandoc` is definitely an improvement. Just had a quick thought on the linebreaking in `locdoc`.

**Comment by takanori-pskq** [comment]
I found a function `ma.doc_note` which solves the wrong indent problem, and the problem mentioned in the docstring of `getdoc`:
>        .. warning::
>          If the function docstring already contained a Notes section, the
>          new docstring will have two Notes sections instead of appending a note
>          to the existing section.
So I think it's better to use `ma.doc_note`.

**Comment by eric-wieser** [comment]
Unfortunately `ma.doc_note` is also broken (#16309), but you're right that using it here is sensible.
Someone can fix `doc_note` in a later PR.

**Comment by eric-wieser** [comment]
Proof that this seems to be working as intended: https://13466-908607-gh.circle-artifacts.com/0/doc/build/html/reference/generated/numpy.ma.vstack.html

**Comment by mattip** [comment]
Close/Reopen to clear CI failure

**Comment by takanori-pskq** [comment]
Now `doc_note` was fixed (#16311), so this change also works correctly.

File: numpy/ma/extras.py [hunk 3]
```diff
@@ -257,10 +252,10 @@ def getdoc(self):
...
         doc = getattr(npfunc, '__doc__', None)
         if doc:
-            sig = self.__name__ + ma.get_object_signature(npfunc)
-            locdoc = "Notes\n-----\nThe function is applied to both the _data"\
-                     " and the _mask, if any."
-            return '\n'.join((sig, doc, locdoc))
+            sig = self.__name__ + ma.get_object_signature(npfunc) + '\n'
+            doc = ma.doc_note(doc, "The function is applied to both the _data"
+                                   " and the _mask, if any.")
+            return '\n'.join((sig, doc))
```
Comment (eric-wieser) [inline numpy/ma/extras.py:258]: No point adding `\n` in two different places:
```suggestion
            sig = self.__name__ + ma.get_object_signature(npfunc)
            doc = ma.doc_note(doc, "The function is applied to both the _data "
                                   "and the _mask, if any.")
            return '\n\n'.join((sig, doc))
```

**Review by eric-wieser — APPROVED** [review]

File: numpy/ma/extras.py [hunk 4]
```diff
@@ -258,9 +253,9 @@ def getdoc(self):
         doc = getattr(npfunc, '__doc__', None)
         if doc:
             sig = self.__name__ + ma.get_object_signature(npfunc)
-            locdoc = "Notes\n-----\nThe function is applied to both the _data"\
-                     " and the _mask, if any."
-            return '\n'.join((sig, doc, locdoc))
+            doc = ma.doc_note(doc, "The function is applied to both the _data "
```
Comment (eric-wieser) [inline numpy/ma/extras.py:256]: I wonder if as a follow-up we can set `self.__wrapped__ = npfunc` to avoid having to add a manual signature line.

**Comment by mattip** [comment]
Thanks @takanori-pskq

**Pull request merged**